src/Symfony/Bundle/FrameworkBundle/Command/DebugAutoconfigurationCommand.php

```diff
@@ -0,0 +1,153 @@
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Dumper\YamlDumper;
+use Symfony\Component\VarDumper\Cloner\VarCloner;
+use Symfony\Component\VarDumper\Dumper\AbstractDumper;
+use Symfony\Component\VarDumper\Dumper\CliDumper;
+
+/**
+ * A console command for autoconfiguration information.
+ *
+ * @internal
+ */
+class DebugAutoconfigurationCommand extends Command
+{
+    protected static $defaultName = 'debug:autoconfiguration';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDefinition(array(
+                new InputArgument('search', InputArgument::OPTIONAL, 'A search filter'),
+                new InputOption('tags', null, InputOption::VALUE_NONE, 'Displays autoconfiguration interfaces/class grouped by tags'),
+            ))
+            ->setDescription('Displays current autoconfiguration for an application')
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> command displays all services that
+are autoconfigured:
+
+  <info>php %command.full_name%</info>
+
+You can also pass a search term to filter the list:
+
+  <info>php %command.full_name% log</info>
+
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $errorIo = $io->getErrorStyle();
+
+        $autoconfiguredInstanceofItems = $this->getContainerBuilder()->getAutoconfiguredInstanceof();
+
+        if ($search = $input->getArgument('search')) {
+            $autoconfiguredInstanceofItems = array_filter($autoconfiguredInstanceofItems, function ($key) use ($search) {
+                return false !== stripos(str_replace('\\', '', $key), $search);
+            }, ARRAY_FILTER_USE_KEY);
+
+            if (empty($autoconfiguredInstanceofItems)) {
+                $errorIo->error(sprintf('No autoconfiguration interface/class found matching "%s"', $search));
+
+                return 1;
+            }
+        }
+
+        ksort($autoconfiguredInstanceofItems, SORT_NATURAL);
+
+        $io->title('Autoconfiguration');
+        if ($search) {
+            $io->text(sprintf('(only showing classes/interfaces matching <comment>%s</comment>)', $search));
+        }
+        $io->newLine();
+
+        /** @var ChildDefinition $autoconfiguredInstanceofItem */
+        foreach ($autoconfiguredInstanceofItems as $key => $autoconfiguredInstanceofItem) {
+            $tags = $autoconfiguredInstanceofItem->getTags();
+            $tableRows = array();
```

Comment:
maybe we can get rid of the `$tableRows` array by using `addRow()` and `render()`, as this is for `master`. IMO its more readable afterwards

docs: https://symfony.com/doc/current/components/console/helpers/table.html#modifying-rendered-tables

Reply:
I'm not sure it's more readable. To have the same style that the others commands, it is necessary to configure the table.

```
diff --git a/src/Symfony/Bundle/FrameworkBundle/Command/DebugAutoconfigurationCommand.php b/src/Symfony/Bundle/FrameworkBundle/Command/DebugAutoconfigurationCommand.php
index bffdd32751..6d80649634 100644
--- a/src/Symfony/Bundle/FrameworkBundle/Command/DebugAutoconfigurationCommand.php
+++ b/src/Symfony/Bundle/FrameworkBundle/Command/DebugAutoconfigurationCommand.php
@@ -12,6 +12,7 @@
 namespace Symfony\Bundle\FrameworkBundle\Command;
 
 use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
 use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputOption;
Stash this hunk [y,n,q,a,d,j,J,g,/,e,?]? y
@@ -91,26 +92,30 @@ EOF
 
         /** @var ChildDefinition $autoconfiguredInstanceofItem */
         foreach ($autoconfiguredInstanceofItems as $key => $autoconfiguredInstanceofItem) {
-            $tableRows = array();
+            $section = $output->section();
+            $table = new Table($section);
+            $table->setStyle(Table::getStyleDefinition('symfony-style-guide'));
+            $table->addRow(array('Option', 'Value'));
 
             foreach ($autoconfiguredInstanceofItem->getTags() as $tag => $tagAttributes) {
-                $tableRows[] = array('Tag', $tag);
+                $table->addRow(array('Tag', $tag));
                 if ($tagAttributes !== array(array())) {
-                    $tableRows[] = array('Tag attribute', $this->dumpTagAttribute($tagAttributes));
+                    $table->addRow(array('Tag attribute', $this->dumpTagAttribute($tagAttributes)));
                 }
             }
 
             if ($autoconfiguredInstanceofItem->getMethodCalls()) {
-                $tableRows[] = array('Method call', $this->dumpMethodCall($autoconfiguredInstanceofItem));
+                $table->addRow(array('Method call', $this->dumpMethodCall($autoconfiguredInstanceofItem)));
             }
 
             if ($autoconfiguredInstanceofItem->getBindings()) {
-                $tableRows[] = array('Bindings', $this->dumpBindings($autoconfiguredInstanceofItem));
+                $table->addRow(array('Bindings', $this->dumpBindings($autoconfiguredInstanceofItem)));
             }
 
             $io->writeln(sprintf("Autoconfiguration for \"%s\"\n==============================================", $key));
             $io->newLine();
-            $io->table(array('Option', 'Value'), $tableRows);
+            $table->render();
+            $io->newLine();
         }
     }

```